module3/labs/week1/HowTheWebWorks.md

```diff
@@ -0,0 +1,121 @@
+
+## Seeing Request/Response in action
+Let's open [Command Prompt](https://www.makeuseof.com/tag/a-beginners-guide-to-the-windows-command-line/) and run some commands to connect to Google's home page.
+
+Enter `curl google.com -v` in a terminal window and review the output.
+
+```
+* Rebuilt URL to: google.com/
+*   Trying 2607:f8b0:400f:800::200e...
+* Connected to google.com (2607:f8b0:400f:800::200e) port 80 (#0)
+> GET / HTTP/1.1
+> Host: google.com
+> User-Agent: curl/7.43.0
+> Accept: */*
+>
+< HTTP/1.1 301 Moved Permanently
+< Location: http://www.google.com/
+< Content-Type: text/html; charset=UTF-8
+< Date: Thu, 31 Aug 2017 01:09:30 GMT
+< Expires: Sat, 30 Sep 2017 01:09:30 GMT
+< Cache-Control: public, max-age=2592000
+< Server: gws
+< Content-Length: 219
+< X-XSS-Protection: 1; mode=block
+< X-Frame-Options: SAMEORIGIN
+<
+<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
+<TITLE>301 Moved</TITLE></HEAD><BODY>
+<H1>301 Moved</H1>
+The document has moved
+<A HREF="http://www.google.com/">here</A>.
+</BODY></HTML>
+* Connection #0 to host google.com left intact
+```
+
+The `>` symbols indicate part of the request being sent to the server. The `<` symbols indicate the response coming back to our system. Notice:
+
+* Request
+    * The request line with a verb, URI path, and HTTP protocol.
+    * The headers providing a host, user agent (identifying the browser or client software), and an indication of the type of response the client will accept.
+    * Empty body.
+* Response
+    * The status line with an HTTP protocol, the status code and reason phrase.
+    * Headers.
+    * A Body with HTML.
+
+Sending a `curl` request to https://google.com will provide the actual site that is displayed when we visit google in a browser.
+
+> Send a `curl` request to a web site of your choice. Note many of the same pieces from the request/response above, with **slightly** more HTML. 
+
+## How do Users Create Requests?
+https://backend.turing.edu/module2/lessons/request_response_anatomy A User doesn't need to know the inner workings of HTTP in order to use a website. They don't need to know they are sending a `POST` request to `/orders`. The requests they send are generated for them by the browser.
+
+Requests are created by:
+
+1. Typing a URL into a browser address bar
+    * Always a `GET` request
+    * Path is hand-typed
+    * Can include Query Params that must be hand-typed
+    * Does not include a body
+1. Clicking Links or Buttons
+    * The developer decides what Verb, Path and Query Params are included
+    * Does not include a body
+1. Form submission
+    * Developer decides on Verb and Path
+    * Can include query params but usually does not.
+    * User inputs data into form fields. This data is included in the body of the request
+
+## Diagram Process
+
+> Imagine you are signing up as a new user of github.  Diagram the request/response cycle for the following sign-up steps:
```

Comment:
I like that the deliverable here is a Diagram! Feels like we might want a bit more hands-on lab here, but it's tough because it's such a conceptual lesson. Maybe searching for requests on websites the use becomes part of the lab?